rust/connlib/tunnel/src/client.rs

```diff
@@ -184,9 +229,16 @@ where
         };
 
         if let Some(dns_packet) = dns::parse(&tunnel.resources.read(), packet.as_immutable()) {
-            if let Err(e) = send_dns_packet(&device_writer, dns_packet) {
-                tracing::error!(err = %e, "failed to send DNS packet");
-                let _ = tunnel.callbacks.on_error(&e.into());
+            match dns_packet {
```

Comment:
Change the `if let` to a `match` to collapse them and reduce indentation.

Reply:
we still need to `send_dns_packet` I don't think we can collapse them.

Reply:
Huh? I think the following should work:

```rust
match dns::parse(&tunnel.resources.read(), packet.as_immutable()) {
	Some(dns::ResolveStrategy::LocalResponse(pkt)) => {
		if let Err(e) = send_dns_packet(&device_writer, pkt) {
			tracing::error!(err = %e, "failed to send DNS packet");
			let _ = tunnel.callbacks.on_error(&e.into());
		}
	}
	Some(dns::ResolveStrategy::ForwardQuery(query)) => {
		tunnel.role_state.lock().dns_query(query);
	}
	None => {}
}
```

Reply:
Hah, I thought you meant the `send_dns_packet` `if..let`. Gotcha.